### ISSUE TITLE
is it a bug while computing contrastive loss?

### ISSUE THREAD

**Issue by gaofei** [mlfoundations/open_clip#1002]
in the func forward of the ClipLoss, while compute the logits of per image/text, i found that  logit_scale * image_features @ text_features.T  doesn't equal to logit_scale * (image_features @ text_features.T), and also logits_per_image is not equal to logits_per_text.T,  doesn't anyone know why?
            logits_per_image = logit_scale * image_features @ text_features.T
            logits_per_text = logit_scale * text_features @ image_features.T

**Comment by rwightman** [comment]
@gaofei no bug that I see, what do you mean by 'equals' there's always float rounding and you can never compare floats outputs to be exactly equal
